e2e-tests/mdx/gatsby-node.js

```diff
@@ -28,4 +42,81 @@ exports.createSchemaCustomization = ({ actions }) => {
       title: String
     }
   `)
-}
\ No newline at end of file
+}
+
+// Add a HTML field for the RSS plugin
+exports.onCreateNode = async ({ node, actions, getNode, reporter, cache }) => {
+  const { createNodeField } = actions
+  if (node.internal.type === `Mdx`) {
+    // Set slug based on frontmatter, fall back to filename
+    const fileNode = getNode(node.parent)
+
+    if (!fileNode) {
+      return
+    }
+
+    const slug = []
+    if (fileNode.relativeDirectory) {
+      slug.push(fileNode.relativeDirectory)
+    }
+
+    if (node.frontmatter.slug) {
+      slug.push(slugify(node.frontmatter.slug))
+    } else if (fileNode.name !== `index`) {
+      slug.push(slugify(fileNode.name))
+    }
+
+    createNodeField({
+      node,
+      name: `slug`,
+      value: `/${slug.join(`/`)}`,
+    })
+
+
+    createNodeField({
+      node,
+      name: `date`,
+      value: fileNode.ctime,
+    })
+
+    // Render as HTML for the RSS feed
+    if (fileNode.sourceInstanceName !== `posts`) {
+      return
+    }
+
+    const result = await compileMDX(
+      {
+        source: node.body,
+        absolutePath: fileNode.absolutePath,
+      },
+      {
+        // These options are requried to allow rendering to string
+        outputFormat: `function-body`,
+        useDynamicImport: true,
+        // providerImportSource: "#",
+        // Add any custom options or plugins here
+      },
+      cache,
+      reporter
+    )
+
+    if (result && result.processedMDX) {
+      const { run } = await import("@mdx-js/mdx")
+      const args = {
+        ...runtime,
+        // useMDXComponents: () => ({ 
+        //   // Example: import("./src/components/example.js")
+        // }),
```

Comment:
I couldn't get the rendering of imported and passed components to work :( 

Importing these components, even with `const Example = await import("...")` will complain about using sync imports.

Also the theme-ui import in `html.mdx` fails:

https://github.com/gatsbyjs/gatsby/pull/36214/files#diff-b75be5bf8876329b1108eabff932aac33f28401a884334fce2648ad3a29e71f5L1-L7

I guess because the imported Message is an unresolved promise?


Any idea how to get these working, without supporting ESM in Gatsby globally?